x/gov/client/rest/util.go

```diff
@@ -0,0 +1,199 @@
+	}
+	return nil
+}
+
+func (req baseReq) baseReqValidate(w http.ResponseWriter) bool {
+	if len(req.Name) == 0 {
+		w.WriteHeader(http.StatusUnauthorized)
+		err := errors.Errorf("Name required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+
+	if len(req.Password) == 0 {
+		w.WriteHeader(http.StatusUnauthorized)
+		err := errors.Errorf("Password required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+
+	// if len(req.ChainID) == 0 {
+	// 	w.WriteHeader(http.StatusUnauthorized)
+	// 	err := errors.Errorf("ChainID required but not specified")
+	// 	w.Write([]byte(err.Error()))
+	// 	return false
+	// }
+
+	if req.AccountNumber < 0 {
+		w.WriteHeader(http.StatusUnauthorized)
+		err := errors.Errorf("Account Number required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+
+	if req.Sequence < 0 {
+		w.WriteHeader(http.StatusUnauthorized)
+		err := errors.Errorf("Sequence required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+	return true
+}
+
+func (req postProposalReq) Validate(w http.ResponseWriter) bool {
+	if len(req.Title) == 0 {
+		w.WriteHeader(http.StatusUnauthorized)
+		err := errors.Errorf("Title required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+
+	if len(req.ProposalType) == 0 {
+		w.WriteHeader(http.StatusUnauthorized)
+		err := errors.Errorf("ProposalType required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+
+	if len(req.Proposer) == 0 {
+		w.WriteHeader(http.StatusUnauthorized)
+		err := errors.Errorf("Proposer required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+
+	if len(req.InitialDeposit) == 0 {
+		w.WriteHeader(http.StatusUnauthorized)
+		err := errors.Errorf("InitialDeposit required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+	return req.BaseReq.baseReqValidate(w)
+}
+
+func (req depositReq) Validate(w http.ResponseWriter) bool {
+	if len(req.Depositer) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		err := errors.Errorf("Depositer required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+
+	if len(req.Amount) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		err := errors.Errorf("Amount required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+	return req.BaseReq.baseReqValidate(w)
+}
+
+func (req voteReq) Validate(w http.ResponseWriter) bool {
+	if len(req.Voter) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		err := errors.Errorf("Voter required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+
+	if len(req.Option) == 0 {
+		w.WriteHeader(http.StatusBadRequest)
+		err := errors.Errorf("Option required but not specified")
+		w.Write([]byte(err.Error()))
+		return false
+	}
+	return req.BaseReq.baseReqValidate(w)
+}
+
+func signAndBuild(w http.ResponseWriter, ctx context.CoreContext, baseReq baseReq, msg sdk.Msg, cdc *wire.Codec) {
```

Comment:
this is a good function which I think we should attempt to build out with a more generic base-request for other golang rest requests (probably should live in `client/lcd` ) ......... maybe add a TODO for this here?